VisualPinball.Unity/VisualPinball.Unity/Import/Material/HdrpMaterialConverter.cs

```diff
@@ -9,23 +9,6 @@ namespace VisualPinball.Unity.Import.Material
 {
 	public class HdrpMaterialConverter : IMaterialConverter
 	{
-		private readonly int SurfaceType = Shader.PropertyToID("_SurfaceType");
-		private readonly int BaseColor = Shader.PropertyToID("_BaseColor");
-		private readonly int BaseColorMap = Shader.PropertyToID("_BaseColorMap");
-
-		private readonly int NormalMap = Shader.PropertyToID("_NormalMap");
-		private readonly int NormalMapSpace = Shader.PropertyToID("_NormalMapSpace");
-		private readonly int NormalScale = Shader.PropertyToID("_NormalScale");
-
-		private readonly int Metallic = Shader.PropertyToID("_Metallic");
-		private readonly int Smoothness = Shader.PropertyToID("_Smoothness");
-		private readonly int BlendMode = Shader.PropertyToID("_BlendMode");
-		private readonly int TransparentSortPriority = Shader.PropertyToID("_TransparentSortPriority");
-		private readonly int AlphaCutoffEnable = Shader.PropertyToID("_AlphaCutoffEnable");
-		private readonly int SrcBlend = Shader.PropertyToID("_SrcBlend");
-		private readonly int DstBlend = Shader.PropertyToID("_DstBlend");
-		private readonly int ZWrite = Shader.PropertyToID("_ZWrite");
```

Comment:
Because they become more and more and the code less readable, i. e. it's better readable with the direct property name. Performance-wise it shouldn't have an effect, since you import a pinball table only once. If it really were a problem, Unity themselves wouldn't do it that way. Just look at StandardsToHDLitMaterialUpgrader.cs, BaseLitGUI.cs, HDLitMasterNode.cs, etc.

Reply:
Okay, I had a look at this. [Here](https://github.com/JetBrains/resharper-unity/wiki/Avoid-using-string-based-names-for-setting-and-getting-properties-on-Animators,-Shaders-and-Materials) is the description of the inspection I'm getting.

As you said yourself, the Unity HDRP code seems somewhat hacky, so that's probably not the benchmark we should take. But I also don't understand your readability arguments. With a cached prop, it looks like that:

```cs
unityMaterial.SetInt(NormalMapSpace, 0);
```

Non-cached, it looks like that:

```cs
unityMaterial.SetInt("_NormalMapSpace", 0);
```

I would actually prefer the former. Of course, the cached version adds a whole block of property definitions, but they can easily be hidden into a `#region`.

I can refactor this quickly if you wish.